fahrschein/src/main/java/org/zalando/fahrschein/PlatformAccessTokenProvider.java

```diff
@@ -0,0 +1,57 @@
+package org.zalando.fahrschein;
+
+import lombok.AllArgsConstructor;
+import org.zalando.fahrschein.AccessTokenProvider;
+import org.zalando.fahrschein.Preconditions;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Default {@link AccessTokenProvider access token provider} built for Zalando's Platform IAM which provides
+ * OAuth2 tokens as files in a mounted directory. Each token is represented as a set of two individual files:
+ *
+ * <dl>
+ *     <dt>{name}-token-type</dt>
+ *     <dd>Contains the token type, e.g. Bearer.</dd>
+ *
+ *     <dt>{name}-token-secret</dt>
+ *     <dd>Contains the actual secret, e.g. a Json Web Token (JWT)</dd>
+ * </dl>
+ *
+ * @see <a href="https://kubernetes-on-aws.readthedocs.io/en/latest/user-guide/zalando-iam.html">Zalando Platform IAM Integration</a>
+ */
+@AllArgsConstructor
```

Comment:
This is a bad example of lombok usage because it becomes rather implicit to the Lombok usage which one of the two strings of the constructor come first, with high potential for mixing them up. 
And what about using `Path directory`?

Reply:
Understood, I was expecting some suggestions over this. The reason why I didn't use `Path` is to reduce the complexity towards our clients, but the constructor mixup could cause serious issues. 👍